### PR TITLE
add typed router configs in examples

### PR DESCRIPTION
**Pull request by justincely**
Signed-off-by: Justin Ely <justincely@gmail.com>

Following discussion in #15017, the router should include the typed config.  

Commit Message: add typed router configs in examples
Additional Description:
Risk Level: 
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:

### CODE REVIEW

**Comment by phlax** [comment]
@justincely im wondering about this - mostly i have been trying to keep the configs as simple as possible

so im wondering if its necessary - at least in all of the examples

**Comment by phlax** [comment]
and whether we can have some kinda default here

**Comment by justincely** [comment]
@phlax I guess that depends on how the discussion plays out in that issue.  @htuch mentioned that a `typed_config` without a type URL is rather invalid.  And the internals will fail these configs through the asserts if you don't do an `opt` build. 

I rather like the idea of defaults personally though

**Comment by phlax** [comment]
> I rather like the idea of defaults personally though

im wondering if we can remove the `typed_config: {}` altogether

**Comment by phlax** [comment]
>  I wouldn't expect that to work

i dont think it does work now - but i think it would be preferable to adding empty or default config - not sure if that makes sense or whether not including at all is possible or would mean something different

**Comment by justincely** [comment]
so, interestingly it seems to work fine.  Just deleted the `typed_config` and the examples I've tried work fine.  Pushing it up now, feel free to test and check.  I did the `gzip` and the `fault_injection` examples.  Config dumps just show an empty config, but server doesn't crash and handles traffic fine.

**Comment by justincely** [comment]
though, neither of us thinking this worked leads me to believe this should be documented better :)

**Comment by phlax** [comment]
> so, interestingly it seems to work fine

wow! theres a lot of these dotted around the code base

```
$ git grep "typed_config: {}"
...
```

> . Just deleted the typed_config and the examples I've tried work fine. Pushing it up now, feel free to test and check. 

if it passes ci - then i think we should remove

> though, neither of us thinking this worked leads me to believe this should be documented better :)

yep!

**Comment by justincely** [comment]
@phlax do you want me to remove all empty typed config?  I just did the router in the examples; since that was the specifics of the discussion, but I can do the others if you think it's prudent?

**Comment by phlax** [comment]
> do you want me to remove all empty typed config?

i think yes - the problem is that if you leave some they inevitably grow and spread through copypasta

also - i think its a better pattern/clearer to remove these so probs best to just remove everywhere

File: test/server/listener_manager_impl_test.cc [hunk 1]
```diff
@@ -4107,7 +4071,7 @@ TEST_F(ListenerManagerImplWithRealFiltersTest, OriginalDstFilterStopsIteration)
     traffic_direction: OUTBOUND
     listener_filters:
     - name: "test.listener.original_dst"
-      typed_config: {}
+      typed_config: {
```
Comment (phlax) [inline test/server/listener_manager_impl_test.cc:4074]: seems like this didnt get deleted properly

**Comment by phlax** [comment]
the one thing im not sure about is removing from test data - probs its best - but if we can get ci to pass ill ping for confirmation

**Comment by justincely** [comment]
sounds good

**Comment by phlax** [comment]
seems like test pass

> the one thing im not sure about is removing from test data

im thinking if tests pass this shouldnt be a problem

**Review by phlax — APPROVED** [review]
lgtm, thanks @justincely

**Pull request merged**